docs/firestore/pagination.md

```diff
@@ -0,0 +1,187 @@
+---
+title: Pagination
+description: Pagination example using Cloud Firestore.
+next: /functions/usage
+previous: /firestore/usage-with-flatlists
+---
+
+Pagination using cloud firestore can be done in various ways but here's the very basic way to do it.
+We are going to make the pagination using querying features provided by firestore.
+[`orderBy`, `limit`, `startAfter`]
+
+# Setup state
+
+First, setup a component which will display the list of data.
+The component will have 2 separate states; `last` and `userData`:
```

Comment:
A bit more compact but most importantly the variable is most correctly named lastDocument in the code so reference it as such

```suggestion
First, create a list display component with 2 state items; `lastDocument` and `userData`:
```